syft/pkg/cataloger/java/graalvm_native_image_cataloger.go

```diff
@@ -0,0 +1,316 @@
+func NewNativeImageCataloger() *NativeImageCataloger {
+	return &NativeImageCataloger{}
+}
+
+// Name returns a string that uniquely describes a native image cataloger
+func (c *NativeImageCataloger) Name() string {
+	return nativeImageCatalogerName
+}
+
+// getPackage returns the package given within a NativeImageComponent.
+func getPackage(component NativeImageComponent) pkg.Package {
+	var cpes []pkg.CPE
+	for _, property := range component.Properties {
+		cpe, err := pkg.NewCPE(property.Value)
+		if err != nil {
+			log.Infof("native-image cataloger: could not parse CPE: %v", err)
+		}
+		cpes = append(cpes, cpe)
+	}
+	return pkg.Package{
+		Name:         component.Name,
+		Version:      component.Version,
+		Language:     pkg.Java,
+		Type:         pkg.GraalVMNativeImagePkg,
+		MetadataType: pkg.JavaMetadataType,
+		FoundBy:      nativeImageCatalogerName,
+		Metadata: pkg.JavaMetadata{
+			PomProperties: &pkg.PomProperties{
+				GroupID: component.Group,
+			},
+		},
+		CPEs: cpes,
+	}
+}
+
+// decompressSbom returns the packages given within a native image executable's SBOM.
+func decompressSbom(databuf []byte, sbomStart uint64, lengthStart uint64) ([]pkg.Package, error) {
+	var pkgs []pkg.Package
+
+	lengthEnd := lengthStart + 8
+	buflen := len(databuf)
+	if lengthEnd > uint64(buflen) {
+		return nil, errors.New("the sbom_length symbol overflows the binary")
+	}
+
+	length := databuf[lengthStart:lengthEnd]
+	p := bytes.NewBuffer(length)
+	var storedLength uint64
+	err := binary.Read(p, binary.LittleEndian, &storedLength)
+	if err != nil {
+		log.Warnf("native-image-cataloger: could not read from binary file")
+		return nil, err
+	}
+	log.Infof("native-image cataloger: length of SBOM %d.", storedLength)
+
+	sbomEnd := sbomStart + storedLength
+	if sbomEnd > uint64(buflen) {
+		return nil, errors.New("the sbom symbol overflows the binary")
+	}
+	sbomCompressed := databuf[sbomStart:sbomEnd]
+	p = bytes.NewBuffer(sbomCompressed)
+	gzreader, err := gzip.NewReader(p)
+	if err != nil {
+		log.Infof("native-image cataloger: could not decompress the SBOM.")
+		return nil, err
+	}
+	output, err := io.ReadAll(gzreader)
+	if err != nil {
+		log.Infof("native-image cataloger: could not read the decompressed SBOM.")
+		return nil, err
+	}
+	var sbomContent NativeImageCycloneDX
+	err = json.Unmarshal(output, &sbomContent)
+	if err != nil {
+		log.Infof("native-image cataloger: could not unmarshal JSON.")
+		return nil, err
+	}
+
+	for _, component := range sbomContent.Components {
+		p := getPackage(component)
+		pkgs = append(pkgs, p)
+	}
+	return pkgs, nil
+}
+
+// fileError logs an error message when an executable cannot be read.
+func fileError(filename string, err error) (NativeImage, error) {
+	// We could not read the file as a binary for the desired platform, but it may still be a native-image executable.
+	log.Infof("native-image cataloger: unable to read executable (file=%q): %v", filename, err)
```

Comment:
probably `log.Debugf`, since we're scanning all executable files and expect some of them not to be ELF or MachO (e.g. executable scripts, etc.)